can/interfaces/neousys/neousys.py

```diff
@@ -0,0 +1,272 @@
+
+    _fields_ = [
+        ("id", c_uint),
+        ("flags", c_ushort),
+        ("extra", c_ubyte),
+        ("len", c_ubyte),
+        ("data", c_ubyte * 8),
+    ]
+
+
+# valid:2~16, sum of the Synchronization, Propagation, and
+#             Phase Buffer 1 segments, measured in time quanta.
+# valid:1~8, the Phase Buffer 2 segment in time quanta.
+# valid:1~4, Resynchronization Jump Width in time quanta
+# valid:1~1023, CAN_CLK divider used to determine time quanta
+class NeousysCanBitClk(Structure):
+    """ C CAN BIT Clock struct """
+
+    _fields_ = [
+        ("syncPropPhase1Seg", c_ushort),
+        ("phase2Seg", c_ushort),
+        ("jumpWidth", c_ushort),
+        ("quantumPrescaler", c_ushort),
+    ]
+
+
+NEOUSYS_CAN_MSG_CALLBACK = CFUNCTYPE(None, POINTER(NeousysCanMsg), c_uint)
+NEOUSYS_CAN_STATUS_CALLBACK = CFUNCTYPE(None, c_uint)
+
+NEOUSYS_CAN_MSG_EXTENDED_ID = 0x0004
+NEOUSYS_CAN_MSG_REMOTE_FRAME = 0x0040
+NEOUSYS_CAN_MSG_DATA_NEW = 0x0080
+NEOUSYS_CAN_MSG_DATA_LOST = 0x0100
+
+NEOUSYS_CAN_MSG_USE_ID_FILTER = 0x00000008
+NEOUSYS_CAN_MSG_USE_DIR_FILTER = (
+    0x00000010 | NEOUSYS_CAN_MSG_USE_ID_FILTER
+)  # only accept the direction specified in the message type
+NEOUSYS_CAN_MSG_USE_EXT_FILTER = (
+    0x00000020 | NEOUSYS_CAN_MSG_USE_ID_FILTER
+)  # filters on only extended identifiers
+
+NEOUSYS_CAN_STATUS_BUS_OFF = 0x00000080
+NEOUSYS_CAN_STATUS_EWARN = (
+    0x00000040  # can controller error level has reached warning level.
+)
+NEOUSYS_CAN_STATUS_EPASS = (
+    0x00000020  # can controller error level has reached error passive level.
+)
+NEOUSYS_CAN_STATUS_LEC_STUFF = 0x00000001  # a bit stuffing error has occurred.
+NEOUSYS_CAN_STATUS_LEC_FORM = 0x00000002  # a formatting error has occurred.
+NEOUSYS_CAN_STATUS_LEC_ACK = 0x00000003  # an acknowledge error has occurred.
+NEOUSYS_CAN_STATUS_LEC_BIT1 = (
+    0x00000004  # the bus remained a bit level of 1 for longer than is allowed.
+)
+NEOUSYS_CAN_STATUS_LEC_BIT0 = (
+    0x00000005  # the bus remained a bit level of 0 for longer than is allowed.
+)
+NEOUSYS_CAN_STATUS_LEC_CRC = 0x00000006  # a crc error has occurred.
+NEOUSYS_CAN_STATUS_LEC_MASK = (
+    0x00000007  # this is the mask for the can last error code (lec).
+)
+
+NEOUSYS_CANLIB = None
+
+try:
+    if platform.system() == "Windows":
+        NEOUSYS_CANLIB = WinDLL("./WDT_DIO.dll")
+    else:
+        NEOUSYS_CANLIB = CDLL("libwdt_dio.so")
+        logger.info("Loaded Neousys WDT_DIO Can driver")
+except OSError as error:
+    logger.info("Cannot Neousys CAN bus dll or share object: %d", format(error))
+    # NEOUSYS_CANLIB = None
+
+
+class NeousysBus(BusABC):
+    """ Neousys CAN bus Class"""
+
+    def __init__(self, channel, device=0, bitrate=500000, **kwargs):
+        """
+        :param channel: channel number
+        :param device: device number
+        :param bitrate: bit rate. Renamed to bitrate in next release.
+        """
+        super(NeousysBus, self).__init__(channel, **kwargs)
```

Comment:
In Python 3, just use `super()`:

```suggestion
        super().__init__(channel, **kwargs)
```